pkg/instrumentation/python.go

```diff
@@ -113,10 +127,13 @@ func injectPythonSDK(pythonSpec v1alpha1.Python, pod corev1.Pod, index int) (cor
 				},
 			}})
 
+		// copy the default glibc based dir if the autoInstrumentationSrc does not exist to not break compat with old images
+		shellCmd := fmt.Sprintf("test -d %[1]s && cp -r %[1]s %[2]s || cp -r %[3]s %[2]s", autoInstrumentationSrc, pythonInstrMountPath, glibcLinuxAutoInstrumentationSrc)
```

Comment:
Honestly, while this makes tests pass and makes life easier for us, I think it's bad for users. If a user sets the annotation to `musl` and provides an image without the artifacts, we *should* fail and let them know.

Reply:
It's fine for me. If we agree on the general approach would it be fine to get a release out with the updated image installing the musl directory and then open a PR with the code making use of it?

Reply:
Yes, that sounds good to me.

Reply:
PR updating the python image here https://github.com/open-telemetry/opentelemetry-operator/pull/3384